auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2021 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import exceptions, models, fields, api, modules, _
+from odoo.addons.auditlog.models.rule import FIELDS_BLACKLIST
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    field_ids = fields.Many2many(
+        "ir.model.fields"
+    )
+    group_ids = fields.Many2many(
+        "res.groups",
+        help="""Groups that will be allowed to see the logged fields, if left empty 
+                default will be all users with a login""",
+    )
```

Comment:
Actually if we do it like this, it's not possible to for example say: for this particular audit log rule eg on `hr.employee`, we will give access to HR managers to see changes to `job_id`, and to HR recruiters only to `contract_date_end` or something. We can just give all listed groups access to all the listed fields, and not differentiate.

Am I right or seeing something wrong? If it is so, then that's something we should change, and IMO it would also make it look clearer - we make a `<group>` called "User Access" which shows a list view that leads to a new model `auditlog.line.access.rule`, and that model would have many "fields_ids / group_ids" combinations, that would each be linked with a particular `ir.rule` that makes that particular access possible.

Reply:
I had made the extra model in my first iteration, still have that branch.
But in this case, in the example you made we can just create:
            1 auditlog rule with   field_ids: ;'job_id'     and group_id: HR_manager
            then another auditlog rule with : field_ids: 'contract_date_end' and group_id:  'HR_recruiters'

this would create two seperate rules for the two different groups. It;s simpler and creates less complex rules.
I think it's clearer than putting everything in a separate model, after making the "extra model" option I saw it was redundant, it seems cleaner to put everything in auditlog_rule.
But Now i see I made a terrible mistake, there is a unique constraint on auditlog.rule. We need the extra model. 
On it.